types/three/src/core/EventDispatcher.d.ts

```diff
@@ -1,15 +1,21 @@
-export interface BaseEvent {
-    type: string;
+/**
+ * The minimal basic Event that can be dispatched by a {@link EventDispatcher<>}.
+ */
+export interface BaseEvent<TEventType extends string = string> {
+    readonly type: TEventType;
```

Comment:
I don't understand why, but when trying to upgrade to r156 types I'm now getting `EventDispatcher<{}>.dispatchEvent<never>(event: never): void` which is impossible to satisfy, so TS won't accept any event type. I don't see where the `never` is coming from - does anyone here know?

Reply:
It's likely coming from a class that's extending `EventDispatcher` that has no defined events. If you have a class that extends `EventDispatcher`, you'll need to specify the events that can be dispatched. For example:

```ts
class SomeObject extends EventDispatcher<{ myEvent: { withProperty: number } }> {
}
```

would allow you to dispatch:

```ts
someObject.dispatchEvent({ type: 'myEvent', withProperty: 5 });
```

See https://github.com/three-types/three-ts-types/pull/369 for more details and let me know if there's a way to improve the types.

Reply:
If you want to allow dispatching any event you can do:

```ts
class AllowAnyEvent extends Object3D<THREE.Object3DEventMap & Record<string, { [key: string]: unknown }>> {}
const allowAnyEventClass = new AllowAnyEvent();
allowAnyEventClass.dispatchEvent({ type: 'any', data: 5 });
```

Reply:
Excellent, thank you! I will give that a shot.